zipline/pipeline/factors/basic.py

```diff
@@ -489,6 +491,38 @@ def compute(self, today, assets, out, returns, annualization_factor):
         out[:] = nanstd(returns, axis=0) * (annualization_factor ** .5)
 
 
+class PeerCount(SingleInputMixin, CustomFactor):
+    """
+    Peer Count of distinct categories in a given classifier.  This factor
+    is returned by the classifier instance method peer_count()
+
+    **Default Inputs:** None
+
+    **Default Window Length:** 1
+    """
+    window_length = 1
+    missing_value = -1
+    dtype = int64_dtype
+
+    def _validate(self):
+        super(PeerCount, self)._validate()
+        if self.window_length != 1:
```

Comment:
Should we also enforce that `inputs` is length 1?

Reply:
I think this is currently enforced because it inherits from SingleInputMixin

Reply:
Oh cool, I missed that.